### ISSUE TITLE
[BUG] .set is hidden by default

### ISSUE THREAD

**Issue by mrvacbob** [compiler-explorer/compiler-explorer#2409]
**Describe the bug**
The `.set` directive is hidden in the asm output by default, which makes 'gcc -fmerge-all-constants' output appear as if there's a compiler bug where it references a2 in this example without defining it. Unchecking "filter directives" did fix it, but I didn't find that until after finding "compile to binary" and wondering how it was working with no definition.

**Expected behavior**
`.set` is load bearing and defines the symbol a2, so it seems like it shouldn't be filtered out.

**Link**
https://gcc.godbolt.org/z/KW8Ms7

**Comment by partouf** [comment]
I can see the issue with this, but I'm not sure what the solution should be. This is a very strange way of annotating the assembly for this.
@mattgodbolt what do you think

**Comment by mattgodbolt** [comment]
Quite an unusual case, but I agree. Ideally the `.set` would be treated as a reference to `a1`, and not filtered out. We'd also probably need to treat `.set` like we do a definition. It _might_ "just work" if we treat `.set` like our other defintiions...

**Comment by mattgodbolt** [comment]
Agreed: I just merged some fixes for `.set` and hoped it might fix this too, but sadly not in this case.

See 3b507a6db399a31b3d6aff70ad2b9be4b2d82c87 but clearly that wasn't enough to fix this.

**Comment by mattgodbolt** [comment]
so in the linked change above, I handled the special `set X,.` and `set X, . + 0`; it ought to be possible to extend the aliasing type that it introduced to support this (sort of) too.

**Comment by mattgodbolt** [comment]
looks good in staging

**Comment by partouf** [comment]
This is now live
